pkgs/by-name/mi/mirrord/package.nix

```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  testers,
+  mirrord,
+  autoPatchelfHook,
+}:
+
+let
+  manifest = lib.importJSON ./manifest.json;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mirrord";
+  version = manifest.version;
+
+  src =
+    let
+      system = stdenv.hostPlatform.system;
+    in
+    fetchurl (manifest.assets.${system} or (throw "Unsupported system: ${system}"));
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isElf [
+    autoPatchelfHook
+  ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isElf [
+    stdenv.cc.cc.lib
+  ];
+
+  installPhase = ''
+    install -D $src $out/bin/mirrord
+  '';
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = mirrord;
+    };
+    updateScript = ./update.py;
+  };
+
+  meta = {
+    description = "Run local processes in the context of Kubernetes environment";
+    homepage = "https://mirrord.dev/";
+    license = lib.licenses.mit;
+    platforms = builtins.attrNames manifest.assets;
+    maintainers = with lib.maintainers; [ aaronjheng ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

Comment:
It shouldn't be *that* difficult to build from source, right?
https://github.com/metalbear-co/mirrord
https://github.com/metalbear-co/mirrord/blob/main/CONTRIBUTING.md

Reply:
I'm not sure. Just want to push forward #310241. I hope to try building from source in the future.